src/sparsezoo/analyze_v2/model_analysis.py

```diff
@@ -15,25 +15,95 @@
 
 from typing import Dict
 
+import onnx
 import yaml
 
-from sparsezoo.analyze_v2.model_validator import ModelAnalysisModel
+from sparsezoo import Model
 from sparsezoo.analyze_v2.node_analysis import NodeAnalysis
+from sparsezoo.analyze_v2.schemas import ModelAnalysisSchema
 from sparsezoo.analyze_v2.summary_analysis import SummaryAnalysis
+from sparsezoo.utils import ONNXGraph, extract_node_id, extract_node_shapes_and_dtypes
 
 
 class ModelAnalysis:
+    """
+    Given summary and node anlysis, obtain the overall analysis of the onnx model.
+
+    From fine-grained detailed analysis (top):
+
+    SparsityAnalysis:       num_counts, num_counts_sparse
+    MemoryAccessAnalysis:   num_mem_access, num_mem_access_sparse
+    QuantizationAnalysis:   num_bits, num_bits_quant
+
+    ParameterAnalysis:      contains SparsityAnalysis, MemoryAccessAnalysis,
+                             QuantizationAnalysis
+    OperationAnalysis:      contains SparsityAnalysis, MemoryAccessAnalysis,
+                             QuantizationAnalysis
+    DistributionAnalysis:   contains SparsityAnalysis, MemoryAccessAnalysis,
+                             QuantizationAnalysis
+
+    NodeAnalysis:           contains ParameterAnalysis, OperationAnalysis,
+                             DistributionAnalysis per node_id
+    SummaryAnalysis         contains the sum of SparsityAnalysis, MemoryAccessAnalysis,
+                             QuantizationAnalysis per grouping
+    ModelAnalysis           contains NodeAnalysis, SummaryAnalysis
+    """
+
     def __init__(
         self, summary_analysis: SummaryAnalysis, node_analyses: Dict[str, NodeAnalysis]
     ):
         self.summary_analysis = summary_analysis
         self.node_analyses = node_analyses
 
     def to_dict(self):
-        return ModelAnalysisModel(
-            summaries=self.summary_analysis.to_dict(),
-            nodes={id: node.to_dict() for id, node in self.node_analyses.items()},
+        summaries = self.summary_analysis.to_dict()
+        nodes = {}
+        for id, node in self.node_analyses.items():
+            if node.to_dict() is not None:
+                nodes[id] = node.to_dict()
+
+        return ModelAnalysisSchema(
+            summaries=summaries,
+            nodes=nodes,
         ).dict()
 
     def to_yaml(self):
         return yaml.dump(self.to_dict())
+
+
+def analyze(path: str) -> "ModelAnalysis":
+    """
+    Entry point to run the model analysis.
+
+    Carries out analysis of the desired onnx model with respect
+    to the weights in the nodes, if exists.
+
+    :param path: .onnx path or stub
+    """
+    if path.endswith(".onnx"):
+        onnx_model = onnx.load(path)
+    else:
+        model = Model(path)
+        onnx_model_path = model.deployment.files[0].path
+        onnx_model = onnx.load(onnx_model_path)
```

Comment:
this should call out to a helper in `sparsezoo/onnx` that loads an onnx model from a path, also we should default to `onnx.load` rather than default to assume zoo